### PR TITLE
Initial prototype

### PR DESCRIPTION
**Pull request by getTobiasNielsen**
A lot of the types are missing, would need to implement those, either in the file or in their respective places.

https://github.com/paypal/react-paypal-js/issues/49

This is an initial prototype.

I was thinking it could be smart to add one of the default spinners I've seen paypal use, or if people don't like it, let people add a prop with a spinner element or some custom react component, if they fancy that instead.

### CODE REVIEW

File: src/components/PayPalHostedFields.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        // billing address here
+ *     </PayPalHostedFields>
+ * ```
+ */
+
+export const PayPalHostedFields: React.FunctionComponent<PayPalHostedFieldsProps> = ({className = "", forceReRender, children, ...hostedFieldsProps}) => {
+    // const hostedFields = useRef<PayPalHostedFieldsComponent | null>(null);
+    const hostedFields = useRef<any | null>(null);
+    const [{ isResolved, options }] = usePayPalScriptReducer();
+    const [initActions, setInitActions] = useState<OnInitActions | null>(null);
```
Comment (gregjopa) [inline src/components/PayPalHostedFields.tsx:51]: This `initActions` code is unique to the Buttons component. We shouldn't need it for HostedFields.

File: src/components/PayPalHostedFields.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+// useEffect hook for rendering the hosted-fields
+useEffect(() => {
+    // verify the sdk script has successfully loaded
+    if (isResolved === false) {
+        return closeHostedFieldsComponent;
+    }
+
+    // Make sure a client token is provided
+    if (options["data-client-token"] === null) {
```
Comment (gregjopa) [inline src/components/PayPalHostedFields.tsx:71]: I'm wondering how this "data-client-token" value would get set in someone's integration code. Can you share an example of what that would look like with the PayPalScriptProvider and this new PayPalHostedFields component? Maybe as a storybook example?

**Comment by gregjopa** [comment]
Thanks for the PR @getTobiasNielsen! I think this is a great start! 💯 I added a couple comments.

File: src/components/PayPalHostedFields.tsx (hunk 2 above)
Comment (getTobiasNielsen) [reply src/components/PayPalHostedFields.tsx:71]: Yeah, this is interesting. Right now, I am using the dispatch to set it, but I imagine going forward, we can make a client id loader, that will take care of fetching an api that is ready to deliver the client-token.

I think having this separate from hosted-fields, will be more helpful. An alternative, is to give the paypal script provider, a route to fetch.

**Pull request closed without merging**